src/Npgsql/NpgsqlConnection.cs

```diff
@@ -201,7 +201,22 @@ void OpenInternal()
             // have Password populated and should not be overwritten.
             if (Password == null)
             {
-                Password = Settings.Password;
+                if (Settings.Password != null)
+                {
+                    Password = Settings.Password;
+                }
+                else if (Settings.Password == null && !IntegratedSecurity)
+                {
+                    // no password was provided. attempt to pull the password from the pgpass file.
+                    var pgPassFile = new PgPassFile(PgPassFile.GetPgPassFilePath());
+
+                    var user = Settings.Username ?? PgPassEntry.PGPASS_WILDCARD;
```

Comment:
If you do this, remember to add the [CanBeNull] attribute on the user parameter of the match function.